NachoSpigot-Server/src/main/java/net/minecraft/server/DedicatedServer.java

```diff
@@ -42,22 +42,24 @@
     public DedicatedServer(joptsimple.OptionSet options) {
         super(options, Proxy.NO_PROXY, DedicatedServer.a);
         // CraftBukkit end
-        Thread thread = new Thread("Server Infinisleeper") {
-            {
-                this.setDaemon(true);
-                this.start();
-            }
+        if (!NachoConfig.disableInfinitSleeperThreadUsage) {
```

Comment:
What is that thread for?

Reply:
Here: https://web.archive.org/web/20150924165427/https://blogs.oracle.com/dholmes/entry/inside_the_hotspot_vm_clocks

Didn't read everything, but if I understand correctly when you do `Thread.sleep` the JVM checks every 10ms if it needs to stop suspending the thread. When you do something like this (called an infinisleeper) the JVM will instead check every millisecond if it needs to stop suspending.

I think I understand why they did this because the server sleeps when it's not processing a tick. So if the server doesn't need to do a lot, it will sleep most of the time and only tick the server 20 times per second. Hence. when you make a spark report, you see Thread.sleep with 95% usage (for me) which lets you know that the server didn't need to do a lot. If you have 300 players online, this will be a LOT lower since each tick takes longer, so less time to sleep.

But, let's say, a tick just got processed. The server now needs to sleep till the next tick. You want ticks to ALWAYS be 20 per second, and it must be synced with the time. But without an infinisleeper, when you do Thread.sleep to wait for the next tick, it can take up to 10ms longer for the thread to come back to life. This causes desyncs, which you don't want.

That's why I also don't really understand why this is a thing now, but I guess that customizability isn't a bad thing 🤷‍♂️